includes/SiteGen/SiteGen.php

```diff
@@ -438,9 +439,21 @@ public static function get_home_pages( $site_description, $content_style, $targe
 			self::cache_sitegen_response( 'homepages', $generated_homepages );
 		}
 
+		$generated_patterns = self::get_sitegen_from_cache( 'generatedPatterns' );
+
+		// check if custom hero patterns needs to be added
+		$site_classification = self::get_sitegen_from_cache( 'siteclassification' );
+		if ( Patterns::check_custom_hero_needed( $site_classification ) ) {
+			// update content structures and generated patterns
+			$custom_structure = Patterns::get_custom_content_structure();
+			foreach ( $generated_content_structures as $home_slug => $structure ) {
+				$generated_content_structures[ $home_slug ] = $custom_structure;
+			}
```

Comment:
Do we need to use AI to generate content structures if we are going to replace them anyway with the hardcoded array?

Reply:
As discussed, the logic to make a call to AI for generating homepages right now depends on whether the content structures are saved in cache or not. We cannot change or hardcode the content structure before that has happened as the response contains the generated patterns as well, which are used to construct the homepages.